coalib/processes/Processing.py

```diff
@@ -290,32 +291,24 @@ def get_file_dict(filename_list, log_printer=None, allow_raw_files=False):
                   '\n'.join(file_dict.keys()))
     return file_dict
 
-
-def filter_raising_callables(it, exception, *args, debug=False, **kwargs):
-    """
-    Filters all callable items inside the given iterator that raise the
-    given exceptions.
-
-    :param it:        The iterator to filter.
-    :param exception: The (tuple of) exception(s) to filter for.
-    :param args:      Positional arguments to pass to the callable.
-    :param kwargs:    Keyword arguments to pass to the callable.
-    """
-    for elem in it:
-        try:
-            yield elem(*args, **kwargs)
-        except exception:
-            if debug:
-                raise
-
+def debug_flag_checker(bear, debug_bears=False):
+    if debug_bears is False:
+        debug_flag = False
+    elif ((str(debug_bears[0]).lower().strip())=='true') or (
+        (bear.__name__) in debug_bears):
+        debug_flag = True
+    else:
+        debug_flag = False
+    return debug_flag
```

Comment:
Do you need this whole stuff? I never see you using those string-branches^^

Reply:
Yes I needed those, previously it was included in the `filter_raisable_callable_function` and because of this function the `debug_flag` will set to `True` to only those bears which are passed to `--debug-bears`

Reply:
With the help of [test cases](https://github.com/RaiVaibhav/coala/pull/5/files#diff-7c16cc0dc5d9e615bbb54239955f7b5eR609), I hoping that I am clearing the point